### PR TITLE
Add ONNX GridSample op support and tests

### PR DESCRIPTION
**Pull request by antimora**
### Checklist

- [x] Confirmed that `cargo run-checks` command has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Partially addresses https://github.com/tracel-ai/burn/issues/4052

### Changes

Implements ONNX GridSample operator import and codegen, adds corresponding node and registry entries, and updates documentation to mark GridSample as supported. Introduces comprehensive ONNX test cases for bilinear and nearest interpolation modes, including Rust and Python test files. Extends tensor and ndarray backends to support grid_sample_2d with configurable interpolation mode, padding mode, and align_corners options. Updates module exports and test generation macros to include grid_sample.

Adds a fully configurable `GridSampleOptions` API for `grid_sample_2d` that matches PyTorch's `torch.nn.functional.grid_sample` interface. This allows users to specify:

- `mode`: Interpolation mode (`Bilinear`, `Nearest`, `Bicubic`)
- `padding_mode`: How to handle out-of-bounds coordinates (`Zeros`, `Border`, `Reflection`)
- `align_corners`: Whether to align corner pixels

Previously, Burn's grid_sample had hardcoded `align_corners=true` and `Border` padding, which didn't match ONNX/PyTorch defaults (`align_corners=false`, `Zeros` padding).

### Testing

- Updated grid_sample tests with various option combinations
- Added ONNX grid_sample tests (`grid_sample.onnx`, `grid_sample_nearest.onnx`)
- Verified compatibility with PyTorch reference values

### Breaking Change

This is a breaking change for users calling `grid_sample_2d` directly. The function now requires a `GridSampleOptions` parameter:

```rust
// Before
tensor.grid_sample_2d(grid);

// After
tensor.grid_sample_2d(grid, GridSampleOptions::default());
```

### CODE REVIEW

File: crates/burn-core/src/lib.rs [hunk 1]
```diff
@@ -111,7 +111,10 @@ pub mod prelude {
         module::Module,
         tensor::{
             Bool, Device, ElementConversion, Float, Int, Shape, SliceArg, Tensor, TensorData,
-            backend::Backend, cast::ToElement, s,
+            backend::Backend,
+            cast::ToElement,
+            ops::{GridSampleOptions, GridSamplePaddingMode, InterpolateMode},
```
Comment (laggui) [inline crates/burn-core/src/lib.rs]: Hmmm I don't think we should add these to the prelude

File: crates/burn-ndarray/src/ops/tensor.rs [hunk 2]
```diff
@@ -515,10 +515,12 @@ where
...
         tensor: FloatTensor<Self>,
         grid: FloatTensor<Self>,
-        method: InterpolateMode,
+        options: GridSampleOptions,
     ) -> FloatTensor<Self> {
         execute_with_float_dtype!((tensor, grid), |tensor, grid| grid_sample_2d(
-            tensor, grid, method
+            tensor,
+            grid,
+            options.clone()
```
Comment (laggui) [inline crates/burn-ndarray/src/ops/tensor.rs]: clone likely not required?

**Review by laggui — APPROVED** [review]
